File: AtCoder/contest322/c.cpp
#include <bits/stdc++.h>
#include <vector>
using namespace std;


int main() {
    // 1. read inputs
    int n, m;
    cin >> n >> m; 
    vector<int> v(m);
    for (int i = 0; i < m; ++i) cin >> v[i];

    // 2. dp
    vector<int> res(n+1);
    vector<bool> fireworks(n+1);
    for (int i = 0; i < m; ++i) fireworks[v[i]] = true;

    int last = n;
    for (int i = n ; i >= 0; --i) {
	if (fireworks[i]) last = i;
	res[i] = last - i;
    }

    // 3. print results
    for (int i = 1; i <= n; ++i) cout << res[i] << endl;

    return 0;
}

File: AtCoder/contest322/c.py
#  !python3 % < AtCoder/contest322/inputs/c1.in;
#  !python3 % < AtCoder/contest322/inputs/c2.in;
#  !python3 % < AtCoder/contest322/inputs/c3.in;

# --- 1. read inputs
n, m = map(int, input().split(' '))
A = list(map(int, input().split(' ')))

# --- 2. DP
res = [0 for _ in range(n+1)]
fireworks = [False for _ in range(n+1)]
for idx in A: fireworks[idx] = True

last = n
for i in range(n, 0, -1):
    if fireworks[i]: last = i
    res[i] = last - i

# --- 3. print res
for r in res[1:]: print(r)
